### PR TITLE
fix(useKeyModifier): `getModifierState` is not a function

### PR DESCRIPTION
**Pull request by okxiaoliang4**
### Description

fixes: #1440

### Additional context

---

### What is the purpose of this pull request? 

- [x] Bug fix
- [ ] New Feature
- [ ] Documentation update
- [ ] Other

### Before submitting the PR, please make sure you do the following

- [x] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [x] Read the [Pull Request Guidelines](https://github.com/vueuse/vueuse/blob/main/packages/guidelines.md).
- [x] Check that there isn't already a PR that solves the problem the same way to avoid creating a duplicate.
- [x] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

**Comment by antfu** [comment]
Not sure. I guess we could have an `isSupported` flag, instead of silently failing in production.

**Comment by antfu** [comment]
Oh I see.

**Comment by okxiaoliang4** [comment]
> Oh I see. I think instead we shouldn't listen to 'mousedown' and 'mouseup' since they have no effects anyway

For example, if you press ctrl when the browser is out of focus, the `mousedown` event can update the state when you click on the page

File: packages/core/useKeyModifier/index.ts [hunk 1]
```diff
@@ -37,7 +37,8 @@ export function useKeyModifier<Initial extends boolean | null>(modifier: KeyModi
   if (document) {
     events.forEach((listenerEvent) => {
       useEventListener(document, listenerEvent, (evt: KeyboardEvent) => {
```
Comment (antfu) [inline packages/core/useKeyModifier/index.ts:39]: KeyboardEvent | MouseEvent

Comment (okxiaoliang4) [reply packages/core/useKeyModifier/index.ts:39]: how about this? `Event & { getModifierState: (modifier: KeyModifier) => boolean }`

Comment (antfu) [reply packages/core/useKeyModifier/index.ts:39]: It's a internal type, that won't affect users so I don't think it need to be that general

Comment (okxiaoliang4) [reply packages/core/useKeyModifier/index.ts:39]: ok

**Pull request merged**